add definition file

lostfictions opened the pull request.

dchest commented on blake2s.d.ts:
```diff
@@ -0,0 +1,22 @@
+export type ByteArray = number[] | Uint8Array;
+
+export default class BLAKE2s {
+  constructor(digestLength: number | undefined, keyOrConfig: ByteArray | BLAKE2sConfig);
```
`keyOrConfig` is optional.

Can a type definition contain multiple declarations for function that accepts different arguments, like normal TypeScript?

```
constructor(digestLength: number?, key?: ByteArray);
constructor(digestLength: number?, config?: BLAKE2sConfig);
```

I wonder if it also can specify the default value as `digestLength = 32`?

lostfictions replied:
Fixed with an overloaded constructor. You can't provide a parameter initializer in defs, unfortunately.

dchest commented on blake2s.d.ts:
```diff
@@ -0,0 +1,22 @@
+
+export default class BLAKE2s {
+  constructor(digestLength: number | undefined, keyOrConfig: ByteArray | BLAKE2sConfig);
+  processBlock(length: number): void;
```
This is actually a private method.

dchest commented on blake2s.d.ts:
```diff
@@ -0,0 +1,22 @@
+  blockLength: number;
+  keyLength: number;
+  saltLength: number;
+  personalizationLength: number;
```
These numbers are `readonly`.

lostfictions replied:
Oops, missed that! I'm honestly not sure how to resolve the incompatibility between ES modules and UMD defs, so I've declared them as both `static readonly` members and const fields in the namespace -- that seems least likely to break something for a consumer?

lostfictions commented:
Fixed, hopefully. I also added a `tsconfig.json` so the definition file would be linted, and to check the defs against the tests file. That said, I didn't bother to fix or suppress any errors in the tests file other than those related to the defs. Hope that helps!
